### PR TITLE
adjust testing framework and update README

### PR DESCRIPTION
**Pull request by guangbochen**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

https://github.com/rancher/harvester/issues/388

**Solution:**

enable and adjust testing  framework and update the README file

**Related Issue:**
https://github.com/rancher/harvester/issues/388

**Test plan:**

N/A

### CODE REVIEW

File: pkg/server/types.go [hunk 1]
```diff
@@ -99,13 +99,18 @@ func Wait(ctx context.Context, config *rest.Config) error {
...
-func (s *HarvesterServer) Start() error {
+func (s *HarvesterServer) Start(listenOpts *dynamiclistener.Config) error {
 	opts := &server.ListenOpts{
 		Secrets: s.steve.Controllers.Core.Secret(),
 		TLSListenerConfig: dynamiclistener.Config{
 			CloseConnOnCertChange: true,
 		},
 	}
+
+	if listenOpts != nil {
```
Comment (gitlawr) [inline pkg/server/types.go:110]: What's the reason for this change?

File: tests/README.md [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
-    - If we want to review testing vm, we can `export KEEP_TESTING_VM=true` to keep it, default is to delete each testing vm after each test case finished.
-  * Run ./script/test-integration or make test-integration.
\ No newline at end of file
+  * Configure appropriate environment variables
+    - set `KUBECONFIG` path and configure `export USE_EXISTING_CLUSTER=true` to use an existing cluster, default to create a local `Kind` cluster.
+        - For the local `Kind` cluster, you can specify `export KIND_IMAGE_MIRROR="http://your-mirror-address:port" ` to speed up pulling images.
+    - configure `export SKIP_HARVESTER_INSTALLATION=true` to skip Harvester installation, default to install a new harvester chart or update the existing one.
+    - set `export DONT_USE_EMULATION=true` if the harvester runtime supports KVM, default to using `QEMU` software emulation.
+    - set `export KEEP_TESTING_CLUSTER=true` if you want to keep the testing cluster, default to delete the local `Kind` cluster.
+    - set `export KEEP_TESTING_RUNTIME=true` if you want to keep the testing runtime, default to delete the harvester runtime.
```
Comment (gitlawr) [inline tests/README.md]: clarify what harvester runtime means, or use a better word

File: tests/integration/api/image_apis_test.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package api
+package api_test
```
Comment (gitlawr) [inline tests/integration/api/image_apis_test.go:1]: Why do we change the package name

File: pkg/server/types.go (hunk 1 above)
Comment (guangbochen) [reply pkg/server/types.go:110]: the default `CloseConnOnCertChange: true` will close the API connection of the integration test since the cert is updating...

File: tests/integration/api/image_apis_test.go (hunk 3 above)
Comment (guangbochen) [reply tests/integration/api/image_apis_test.go:1]: prefer to following the best practice from https://onsi.github.io/ginkgo/
> Go allows us to specify the books_test package alongside the books package. Using books_test instead of books allows us to respect the encapsulation of the books package: your tests will need to import books and access it from the outside, like any other package. This is preferred to reaching into the package and testing its internals and leads to more behavioral tests. You can, of course, opt-out of this – just change package books_test to package books

**Comment by guangbochen** [comment]
@gitlawr updated, thanks.

**Review by gitlawr — APPROVED** [review]
LGTM
